common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -277,29 +277,30 @@ public void onPermissionRequest(final PermissionRequest request) {
                 // We can only grant or deny permissions for video capture/camera.
                 // To avoid unintentionally granting requests for not defined permissions
                 // we check if the request is for camera.
-                if (!isPermissionRequestForCamera(request)) {
+                if (!CameraPermissionRequest.isValidRequest(request)) {
                     Logger.warn(methodTag, "Permission request is not for camera.");
                     request.deny();
                     return;
                 }
                 // There is a issue in ESTS UX where it sends multiple camera permission requests.
                 // So, if there is already a camera permission request in progress we handle it here.
                 if (mCameraPermissionRequest != null) {
-                    handleRepeatedRequests(request);
+                    Logger.info(methodTag, "Repeated request, granted? " + mCameraPermissionRequest.isGranted());
+                    handleRepeatedCameraRequests(request);
                     return;
                 }
                 Logger.info(methodTag, "New camera request.");
-                mCameraPermissionRequest = request;
-                if (isCameraPermissionGranted()) {
+                mCameraPermissionRequest = new CameraPermissionRequest(request);
+                // The current camera behavior is based on the principle of only prompt once and prompt always.
+                // If the OS level permission is not granted, we show the system prompt.
+                // If the OS level permission was granted previously, we show the rationale to confirm the consent witth the current user.
```

Comment:
nit:

- If the OS level permission was granted previously, we show the rationale to confirm the consent witth the current user.
- Otherwise, show the system prompt.